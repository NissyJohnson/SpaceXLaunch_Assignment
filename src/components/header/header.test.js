import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Header from "./header";

configure({ adapter: new Adapter() });

describe("Header Component: Visual", () => {
  const props = {
    heading: "SpaceX Launch Program",
  };

  it("should render a heading", () => {
    const component = shallow(<Header {...props} />);
    const heading = component.find('[data-testid="header"]').text();
    expect(heading).toBe("SpaceX Launch Program");
  });
});

describe("Have props", () => {
  const props = {
    heading: "SpaceX Launch Program",
  };
  test("Should render the <Header />", () => {
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.exists()).toBe(true);
  });

  it("Should render with props", () => {
    const component = shallow(<Header {...props} />);
    const heading = component.find('[data-testid="header"]').text();
    expect(heading).toBe("SpaceX Launch Program");
  });
});
describe("Have no props", () => {
  it("Should render with props", () => {
    const component = shallow(<Header />);
    const heading = component.find('[data-testid="header"]').text();
    expect(heading).toBe("");
  });
});
