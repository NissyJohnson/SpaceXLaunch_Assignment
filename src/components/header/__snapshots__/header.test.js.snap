// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Have props Should render the <Header /> 1`] = `"<header data-testid=\\"header\\">SpaceX Launch Program</header>"`;
